packages/renderer/src/lib/image/BuildImageFromContainerfile.svelte

```diff
@@ -133,13 +133,34 @@ async function buildContainerImage(): Promise<void> {
   }
 }
 
+export function getRelativePath(context: string, file: string): string {
+  let relativePath = '';
+  // split paths based on backward or forward slashes (works for both Windows and Mac/Linux path formats)
+  let contextPath = context.split(/[\\/]/).filter(dir => dir);
+  let filePath = file.split(/[\\/]/).filter(dir => dir);
+  let index = 0;
+  // find the first place where both paths don't match
+  while (index < contextPath.length && index < filePath.length && contextPath[index] === filePath[index]) {
+    index++;
+  }
+  // add .. for the remaining length of the context path (if the context path didn't fully match part of the file path)
+  // then add the remaining path of the file
+  if (index < contextPath.length) {
+    let back = Array(contextPath.length - index).fill('..');
+    relativePath = `${back.join('/')}/${filePath.slice(index).join('/')}`;
+  } else {
+    relativePath = filePath.slice(index).join('/');
+  }
+  return relativePath;
+}
```

Comment:
There is a method `path.relative(from, to)` in the `path` module, which can't be imported on the renderer side, so if this path handling will be moved to the backend, then maybe that method could be used there.

Reply:
yes, might be more convenient to delegate something on the backend to handle this use-case

Reply:
I'm wondering if that would be interesting to change the signature of the `window.buildImage` function, to accept an absolute path instead of the relative path it accepts currently

Reply:
we can change everything on `window.buildImage` as it's internal

Reply:
but what is sure, it's that we should avoid any path handling on the frontend side while we have all API on backend side

Reply:
Yes, I agree with this. 
But having the frontend compute the relative path seems weird to me with this change, as the containerfile is not expected to be inside the context anymore.

It seems more logical to pass the two paths as absolute, and let the backend compute the relative path (as it is now an implementation detail)

Reply:
finally, I added a `window.pathRelative` function, to keep the same signatures for both `buildImage` functions (internal and API)